src/Forms/DateField.php

```diff
@@ -243,59 +251,38 @@ protected function getFormatter()
      */
     protected function getISO8601Formatter()
     {
+        $locale = i18n::config()->uninherited('default_locale');
         $formatter = IntlDateFormatter::create(
             i18n::config()->uninherited('default_locale'),
             IntlDateFormatter::MEDIUM,
             IntlDateFormatter::NONE
         );
         $formatter->setLenient(false);
-        // CLDR iso8601 date.
+        // CLDR ISO 8601 date.
         $formatter->setPattern('y-MM-dd');
         return $formatter;
     }
 
     public function FieldHolder($properties = array())
     {
-        return $this->renderWithClientView(function () use ($properties) {
-            return parent::FieldHolder($properties);
-        });
-    }
-
-    public function SmallFieldHolder($properties = array())
-    {
-        return $this->renderWithClientView(function () use ($properties) {
-            return parent::SmallFieldHolder($properties);
-        });
-    }
-
-    /**
-     * Generate field with client view enabled
-     *
-     * @param callable $callback
-     * @return string
-     */
-    protected function renderWithClientView($callback)
-    {
-        $clientView = null;
-        if ($this->getShowCalendar()) {
-            $clientView = $this->getClientView();
-            $clientView->onBeforeRender();
+        if ($this->getHTML5()) {
+            // Browsers expect ISO 8601 dates, localisation is handled on the client
+            $this->setDateFormat('y-MM-dd');
```

Comment:
Instead of modifying the state on render, can we move this logic to getDateFormat()? I realise we violate this in many places, but we should probably make fields immutable during render.